tests/integration/targets/mso_schema_template_vrf/tasks/main.yml

```diff
@@ -515,4 +521,55 @@
   assert:
     that:
     - nm_query_change_vrf_contract_again is not changed
-    - nm_query_change_vrf_contract_again.current | length == 2
\ No newline at end of file
+    - nm_query_change_vrf_contract_again.current | length == 2
+
+#  Verifying preferred Endpoint Group for VRF.
+- name: Add VRF preferred group (normal mode)
+  mso_schema_template_vrf:
+    <<: *mso_info
+    schema: '{{ mso_schema | default("ansible_test") }}'
+    template: Template 2
+    vrf: vrf_preferred_group
+    preferred_group: true
+    state: present
+  register: vrf_preferred_group_nm
+
+- name: Verify vrf_preferred_group_nm
+  assert:
+    that:
+    - vrf_preferred_group_nm is changed
+    - vrf_preferred_group_nm.current.name == 'vrf_preferred_group'
+    - vrf_preferred_group_nm.current.displayName == 'vrf_preferred_group'
+    - vrf_preferred_group_nm.current.preferredGroup == True
+
+# ADD BD
```

Comment:
Are the following tasks added just for code coverage?

Reply:
Yes and no, the below tasks shows a particular scenario - you can select the preferred group option in the VRF but to enable it you should also make changes in the EPG.

Reply:
Since this is not an example, it would be good to verify those with assertions

Reply:
removed this part as discussed